vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -169,19 +170,29 @@ def process_weights_after_loading(self, layer: Module) -> None:
             layer.weight_scale = Parameter(weight_scale, requires_grad=False)
             layer.input_scale = None
 
-        # If checkpoint is fp8, requantize the separately quantized logical
-        # weights into a single fp8 weight with a single weight scale.
+        # If checkpoint is fp8, handle that there are N scales for N
+        # shards in a fused module
         else:
-            # Dequant -> Quant with max scale.
-            max_w_scale, weight = requantize_with_max_scale(
-                weight=layer.weight,
-                weight_scale=layer.weight_scale,
-                logical_widths=layer.logical_widths,
-            )
+            # If using marlin (w8a16), kernel uses channelwise weights,
```

Comment:
since we extend to channelwise anyways, figured we might as well not requantize here

Reply:
How does it expand the scales? 

If it is `layer.weight_scale = [A, B, C]` will it expand to `weight_scale = [A, A, ..., B, B, ..., C, C, ...]`?